packages/devtools_server/lib/src/server.dart

```diff
@@ -200,41 +202,51 @@ Future<void> registerLaunchDevToolsService(
-      }
+      try {
+        final uriParams = <String, dynamic>{};
+
+        // Copy over queryParams passed by the client
+        if (params != null) {
+          params['queryParams']
+              ?.forEach((key, value) => uriParams[key] = value);
+        }
+
+        // Add the URI to the VM service
+        uriParams['uri'] = vmServiceUri.toString();
+
+        final devToolsUri = Uri.parse(devToolsUrl);
+        final uriToLaunch = devToolsUri.replace(
+          // If path is empty, we generate 'http://foo:8000?uri=' (missing `/`) and
+          // ChromeOS fails to detect that it's a port that's tunneled, and will
+          // quietly replace the IP with "penguin.linux.test". This is not valid
+          // for us since the server isn't bound to the containers IP (it's bound
+          // to the containers loopback IP).
+          path: devToolsUri.path.isEmpty ? '/' : devToolsUri.path,
+          queryParameters: uriParams,
+        );
 
-      // Add the URI to the VM service
-      uriParams['uri'] = vmServiceUri.toString();
-
-      final devToolsUri = Uri.parse(devToolsUrl);
-      final uriToLaunch = devToolsUri.replace(
-        // If path is empty, we generate 'http://foo:8000?uri=' (missing `/`) and
-        // ChromeOS fails to detect that it's a port that's tunneled, and will
-        // quietly replace the IP with "penguin.linux.test". This is not valid
-        // for us since the server isn't bound to the containers IP (it's bound
-        // to the containers loopback IP).
-        path: devToolsUri.path.isEmpty ? '/' : devToolsUri.path,
-        queryParameters: uriParams,
-      );
-
-      // TODO(dantup): When ChromeOS has support for tunneling all ports we
-      // can change this to always use the native browser for ChromeOS
-      // and may wish to handle this inside `browser_launcher`.
-      //   https://crbug.com/848063
-      final useNativeBrowser = _isChromeOS &&
-          _isAccessibleToChromeOSNativeBrowser(Uri.parse(devToolsUrl)) &&
-          _isAccessibleToChromeOSNativeBrowser(vmServiceUri);
-      if (useNativeBrowser) {
-        await Process.start('x-www-browser', [uriToLaunch.toString()]);
-      } else {
-        await Chrome.start([uriToLaunch.toString()]);
+        // TODO(dantup): When ChromeOS has support for tunneling all ports we
+        // can change this to always use the native browser for ChromeOS
+        // and may wish to handle this inside `browser_launcher`.
+        //   https://crbug.com/848063
+        final useNativeBrowser = _isChromeOS &&
+            _isAccessibleToChromeOSNativeBrowser(Uri.parse(devToolsUrl)) &&
+            _isAccessibleToChromeOSNativeBrowser(vmServiceUri);
+        if (useNativeBrowser) {
+          await Process.start('x-www-browser', [uriToLaunch.toString()]);
+        } else {
+          await Chrome.start([uriToLaunch.toString()]);
+        }
+
+        return {'result': Success().toJson()};
+      } catch (e, s) {
+        return {
```

Comment:
Add a comment explaining why this exact format is used.

Reply:
Done!